src/functions/env.ts

```diff
@@ -0,0 +1,208 @@
+import * as clc from "cli-color";
+
+import { FirebaseError } from "../error";
+import * as envstore from "./envstore";
+
+// Exported for testing-only
+export const ENVSTORE_ID = "firebase-functions";
+const RESERVED_KEY_NAMES = [
+  // Cloud Functions for Firebase
+  "FIREBASE_CONFIG",
+  // Cloud Functions:
+  //   https://cloud.google.com/functions/docs/env-var#best_practices_and_reserved_environment_variables
+  "FUNCTION_TARGET",
+  "FUNCTION_SIGNATURE_TYPE",
+  "K_SERVICE",
+  "K_REVISION",
+  "PORT",
+  // Cloud Run:
+  //   https://cloud.google.com/run/docs/reference/container-contract#env-vars
+  "K_CONFIGURATION",
+];
+
+/**
+ * Format environment variables into console-friendly strings.
+ *
+ * @param {Record<string, string>} envs Environment variables to format.
+ * @return {string} Formatted string suitable for printing.
+ */
+export function formatEnv(envs: Record<string, string>): string {
```

Comment:
After reading Google TS style guide advising against `.forEach`, I think I sided with using a loop instead of using `maps` and it's friends. Seems more readable to use `map `here.